tests/utility/hash_test.cu

```diff
@@ -101,28 +112,38 @@ TEST_CASE("Test cuco::xxhash_64", "")
 template <typename OutputIter>
 __global__ void check_hash_result_kernel_32(OutputIter result)
 {
-  result[0] = check_hash_result<cuco::xxhash_32<int32_t>>(0, 0, 148298089);
-  result[1] = check_hash_result<cuco::xxhash_32<int32_t>>(0, 42, 2132181312);
-  result[2] = check_hash_result<cuco::xxhash_32<int32_t>>(42, 0, 1161967057);
-  result[3] = check_hash_result<cuco::xxhash_32<int32_t>>(123456789, 0, 2987034094);
+  int i = 0;
+
+  result[i++] = check_hash_result<cuco::xxhash_32<char>>(0, 0, 3479547966);
+  result[i++] = check_hash_result<cuco::xxhash_32<char>>(42, 0, 3774771295);
+  result[i++] = check_hash_result<cuco::xxhash_32<char>>(0, 42, 2099223482);
 
-  result[4] = check_hash_result<cuco::xxhash_32<int64_t>>(0, 0, 3736311059);
-  result[5] = check_hash_result<cuco::xxhash_32<int64_t>>(0, 42, 1076387279);
-  result[6] = check_hash_result<cuco::xxhash_32<int64_t>>(42, 0, 2332451213);
-  result[7] = check_hash_result<cuco::xxhash_32<int64_t>>(123456789, 0, 1561711919);
+  result[i++] = check_hash_result<cuco::xxhash_32<int32_t>>(0, 0, 148298089);
+  result[i++] = check_hash_result<cuco::xxhash_32<int32_t>>(0, 42, 2132181312);
+  result[i++] = check_hash_result<cuco::xxhash_32<int32_t>>(42, 0, 1161967057);
+  result[i++] = check_hash_result<cuco::xxhash_32<int32_t>>(123456789, 0, 2987034094);
+
+  result[i++] = check_hash_result<cuco::xxhash_32<int64_t>>(0, 0, 3736311059);
+  result[i++] = check_hash_result<cuco::xxhash_32<int64_t>>(0, 42, 1076387279);
+  result[i++] = check_hash_result<cuco::xxhash_32<int64_t>>(42, 0, 2332451213);
+  result[i++] = check_hash_result<cuco::xxhash_32<int64_t>>(123456789, 0, 1561711919);
 
 #if defined(CUCO_HAS_INT128)
-  result[8] = check_hash_result<cuco::xxhash_32<__int128>>(123456789, 0, 1846633701);
+  result[i++] = check_hash_result<cuco::xxhash_32<__int128>>(123456789, 0, 1846633701);
 #endif
 
-  result[9] = check_hash_result<cuco::xxhash_32<large_key<32>>>(123456789, 0, 3715432378);
+  result[i++] = check_hash_result<cuco::xxhash_32<large_key<32>>>(123456789, 0, 3715432378);
```

Comment:
@sleeepyjack can you please remind me why we are not using a global array to avoid duplications?

Reply:
Agreed, this is a bad solution. What does your proposal look like?

Reply:
https://godbolt.org/z/j58fEbeva would this work?

Reply:
Result from offline discussion: move to backlog for now